### PR TITLE
Fix pdf2txt --boxes-flow=disabled

### PR DESCRIPTION
**Pull request by 0xabu**
This fixes:
```
$ pdf2txt.py --boxes-flow=disabled test.pdf
Traceback (most recent call last):
  File "tools/pdf2txt.py", line 204, in <module>
    sys.exit(main())
  File "tools/pdf2txt.py", line 198, in main
    outfp = extract_text(**vars(A))
  File "tools/pdf2txt.py", line 66, in extract_text
    pdfminer.high_level.extract_text_to_fp(fp, **locals())
  File "pdfminer/high_level.py", line 85, in extract_text_to_fp
    interpreter.process_page(page)
  File "pdfminer/pdfinterp.py", line 896, in process_page
    self.device.end_page(page)
  File "pdfminer/converter.py", line 51, in end_page
    self.cur_item.analyze(self.laparams)
  File "pdfminer/layout.py", line 822, in analyze
    group.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 577, in analyze
    self._objs.sort(
  File "pdfminer/layout.py", line 578, in <lambda>
    key=lambda obj: (1 - laparams.boxes_flow) * obj.x0
TypeError: unsupported operand type(s) for -: 'int' and 'str'
```

Related: Issue #477, PR #479

**How Has This Been Tested?**

I ran it once, and verified that (a) there was no crash, and (b) the alternate layout algorithm was used in `LTLayoutContainer.analyze`.

**Checklist**

This is a trivial fix to existing code. I think most of this is irrelevant. We don't seem to have tests for pdf2txt.

### CODE REVIEW

File: tools/pdf2txt.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 def float_or_disabled(x):
     if x.lower().strip() == "disabled":
-        return x
+        return "disabled"
```
Comment (0xabu) [inline tools/pdf2txt.py:21]: Based on the line above, the intended behaviour appears to have been that "Disabled" etc. is permitted; in that case makes sense to normalise the various cases to a single string.

**Review by jstockwin — APPROVED** [review]
I guess I introduced this one :) 

This PR LGTM, thanks!

One question, though - if we simply returned `None` from `float_or_disable` would we achieve the same thing with less code complexity? (Essentially, use `None` to represent `disabled`).

IIRC the only reason I used `disabled` as a string was because you didn't seem to be able to pass `None` directly from the CLI tool.

**Comment by 0xabu** [comment]
> One question, though - if we simply returned `None` from `float_or_disable` would we achieve the same thing with less code complexity? (Essentially, use `None` to represent `disabled`).

That was the first thing I tried :). The problem is this logic in `extract_text` where we'd skip the `setattr` and be left with the default:
```
        for param in ("all_texts", "detect_vertical", "word_margin",
                      "char_margin", "line_margin", "boxes_flow"):
            paramv = locals().get(param, None)
            if paramv is not None:
                setattr(laparams, param, paramv)
```

IMO it's overly elaborate, and it would be cleaner to construct the LAParams object closer to the argument parser [more like this](https://github.com/0xabu/pdfannots/commit/cb51a716f27c27c5690a2d74d0bfbb14b292d1a5), but I wasn't sure who else might be relying on `extract_text`, so this was the least-invasive change.

**Comment by jstockwin** [comment]
Thanks for the explanation - I agree with the reasoning.

Could you add an entry to the `CHANGELOG.md`, please?

After that, this LGTM and it's down to @pietermarsman for a final review and merge.

**Review by jstockwin — APPROVED** [review]

**Comment by 0xabu** [comment]
Note: this PR is superseded by #682. If you plan to take #682, we can drop this minimal fix.

**Comment by pietermarsman** [comment]
Closing in favor of #682. 

@jstockwin thanks for the review.

@0xabu thanks for fixing it.

**Pull request closed without merging**